src/management-system-v2/lib/engines/deployment.ts

```diff
@@ -0,0 +1,224 @@
+import {
+  getElementMachineMapping,
+  getProcessConstraints,
+  getStartEvents,
+  getTaskConstraintMapping,
+  toBpmnObject,
+} from '@proceed/bpmn-helper';
+// TODO: remove this ignore once the decider is typed
+// @ts-ignore
+// import decider from '@proceed/decider';
+import { Machine, getMachines } from './machines';
+import * as endpoints from './endpoints';
+import { prepareExport } from '../process-export/export-preparation';
+import { Prettify } from '../typescript-utils';
+
+// TODO: better error handling
+
+type ProcessesExportData = Prettify<Awaited<ReturnType<typeof prepareExport>>>;
+
+async function deployProcessToMachines(
+  machines: Machine[],
+  processesExportData: ProcessesExportData,
+) {
+  try {
+    // TODO: check if the order of the processes matters
+    const allMachineRequests = machines.map((machine) => {
+      return Promise.all(
+        processesExportData!.map(async (exportData) => {
+          const version = Object.values(exportData.versions)[0];
+          await endpoints.deployProcess(machine, version.bpmn);
+
+          const userTasks = exportData.userTasks.map((userTask) =>
+            endpoints.sendUserTaskHTML(
+              machine,
+              exportData.definitionId,
+              userTask.filename,
+              userTask.html,
+            ),
+          );
+
+          const images = exportData.images.map((image) =>
+            endpoints.sendImage(machine, exportData.definitionId, image.filename, image.data),
+          );
+
+          await Promise.all([userTasks, images]);
```

Comment:
Does this actually work? It looks like you are running Promise.all on an array that contains two arrays which contain promises instead of of an array of promises. Maybe it should be:
`await Promise.all([...userTasks, ...images])`

Reply:
You're right, I pushed a fix in the next PR